compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala

```diff
@@ -406,15 +412,45 @@ trait GenJSExports extends SubComponent { self: GenJSCode =>
 
         val altsByTypeTest = groupByWithoutHashCode(alts) {
           case ExportedSymbol(alt) =>
-            // get parameter type while resolving repeated params
+            // See the comment in genPrimitiveJSArgs for a rationale about this
+
             val tpe = enteringPhase(currentRun.uncurryPhase) {
```

Comment:
This computation has grown very large. Consider extracting the computation of this `tpe` in a separate method.